### PR TITLE
PERF: improve label speed for bool input by using the scipy.ndimage implementation

### LINKED ISSUE #4833: measure.label speed
This is triggered by [this Stackoverflow question](https://stackoverflow.com/questions/62804953/performance-differences-between-bwlabeln-on-matlab-and-skimage-measure-label-on/62842582#62842582). When I have large binary arrays to label and performance issues, I usually resort to calling the ndimage version. Could we imagine having a `fast_binary` flag which would call the ndimage function? A factor of 3-4 (from a few tests I just ran) is not bad...

### PR DESCRIPTION
**Pull request by sciunto**
## Description

closes #4833 

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/measure/_label.py [hunk 1]
```diff
@@ -1,6 +1,32 @@
+from scipy import ndimage
 from ._ccomp import label_cython as clabel
 
 
+def _label_bool(image, background=None, return_num=False, connectivity=None):
+    """
+    Faster implementation of clabel for boolean input.
+    """
+    from ..morphology._util import _resolve_neighborhood
```
Comment (rfezzani) [inline skimage/measure/_label.py:9]: :facepalm: again the circular import...
Not for this PR, but we may be should move `morphology._util` to `skimage.utils`.

**Review by rfezzani — APPROVED** [review]
Thank you @sciunto.

File: skimage/measure/_label.py (hunk 1 above)
Comment (sciunto) [reply skimage/measure/_label.py:9]: indeed :)

File: skimage/measure/_label.py [hunk 2]
```diff
@@ -1,6 +1,32 @@
...
+    if background == 1:
+        image = ~image
+
+    if connectivity is None:
+        connectivity = image.ndim
+
+    if not 1 <= connectivity <= image.ndim:
+        raise ValueError(
+            "Connectivity below 1 or above %d is illegal."
+            % image.ndim)
```
Comment (jni) [inline skimage/measure/_label.py:19]:
```suggestion
        raise ValueError(
            f'Connectivity for {image.ndim}D image should be in [1, ..., {image.ndim}]. '
            f'Got {connectivity}.'
        )
```

**Review by jni — APPROVED** [review]
@sciunto ideally, we would accompany this PR with a benchmark for label with either bool or integer inputs, and the improvement would be documented in the PR comments. =)

File: skimage/measure/_label.py (hunk 2 above)
Comment (sciunto) [reply skimage/measure/_label.py:19]: Copied from the cython code, I'll fix it there as well. thanks.

**Comment by sciunto** [comment]
must be OK now.

**Comment by hmaarrfk** [comment]
Sigh, i wish there was a "merge when passing" button.

**Comment by jni** [comment]
@hmaarrfk We've got you covered! 😂 

In all seriousness, a great UI suggestion.

**Comment by hmaarrfk** [comment]
> In all seriousness, a great UI suggestion.

Gitlab has it. it has been a blessing. Except for the fact that CIs still take forever for small changes..

**Pull request merged**